common/src/main/java/dev/schmarrn/lighty/mode/CarpetMode.java

```diff
@@ -91,42 +82,58 @@ public void compute(ClientLevel world, BlockPos pos, BufferBuilder builder) {
             // Carpet is just one pixel high
             offset = 1f / 16f;
         }
-
-        matrixStack.pushPose();
-        matrixStack.translate(posUp.getX(), (posUp.getY()) + offset, posUp.getZ());
-        Minecraft.getInstance().getBlockRenderer().renderBatched(
-                overlayState,
-                posUp,
-                world,
-                matrixStack,
-                builder,
-                false,
-                random
-        );
-        matrixStack.popPose();
+        double x = pos.getX();
+        double y = pos.getY() + 1 + offset;
+        double z = pos.getZ();
+        //TOP
+        builder.vertex(x, y + 1 / 16f, z).uv(0, 0).color(color).endVertex();
+        builder.vertex(x, y + 1 / 16f, z + 1).uv(0, 1).color(color).endVertex();
+        builder.vertex(x + 1, y + 1 / 16f, z + 1).uv(1, 1).color(color).endVertex();
+        builder.vertex(x + 1, y + 1 / 16f, z).uv(1, 0).color(color).endVertex();
+        if (offset > 0.001f)
+            pos = pos.above();
```

Comment:
please put {} and a comment something like "needed for face culling functionality below"